webpages/settings/index.js

```diff
@@ -169,6 +169,15 @@ const vue = new Vue({
       });
       console.log("Updated", this.addonSettings[addon._addonId]);
     },
+    loadPreset(id, addon) {
+      if (window.confirm("Are you sure you want to load this preset?")) {
+        var preset = addon.presets.filter((item) => item.id == id)[0];
```

Comment:
You can use `Array.prototype.find`, but I'd prefer if this function just received the preset object itself.